decidim-debates/app/commands/decidim/debates/close_debate.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Debates
+    # A command with all the business logic when a user updates a debate.
+    class CloseDebate < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form         - A form object with the params.
+      def initialize(form)
+        @form = form
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the debate.
+      # - :invalid if the form wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) if form.invalid?
+
+        close_debate
+        broadcast(:ok, debate)
+      end
+
+      private
+
+      attr_reader :form
+
+      def close_debate
+        @debate = Decidim.traceability.perform_action!(
+          :close,
+          form.debate,
+          form.current_user
+        ) do
+          form.debate.update!(attributes)
+        end
+
+        Decidim::EventsManager.publish(
+          event: "decidim.events.debates.debate_closed",
+          event_class: Decidim::Debates::CloseDebateEvent,
+          resource: debate,
+          followers: debate.followers
+        )
+      end
+
+      def attributes
+        {
+          conclusions: {
+            I18n.locale => form.conclusions
```

Comment:
This is different from what the same command in admin is doing.
The admin command is not using a `conclusions: {locale => conclusions}` hash, but instead directly assigns `conclusions: conslusions`, some of both is wrong no?

On the other side, it may be good to extract the logic into a concern?

Reply:
Ok, I see, the admin form already returns a hash with all translations while the public form only contains the conclusions in the participant's locale

Reply:
Yup, since the users only write in a single language we need to fake the hash somehow, admin in the other hand can write in different languages. I'll look into the concern part.

Reply:
I've tried some solutions but it doesn't seem worth it to have to add a concern for just a simple method, I've left it as it is.